core/src/main/resources/jenkins/formelementpath/form-element-path.js

```diff
@@ -0,0 +1,196 @@
+                    break;
+                case "file":
+                    // to support structured form submission with file uploads,
+                    // rename form field names to unique ones, and leave this name mapping information
+                    // in JSON. this behavior is backward incompatible, so only do
+                    // this when
+                    p = findParent(e);
+                    if (e.getAttribute("jsonAware") != null) {
+                        var on = e.getAttribute("originalName");
+                        if (on != null) {
+                            addProperty(p, on, e);
+                        } else {
+                            addProperty(p, e.name, e);
+                        }
+                    }
+                    break;
+                // otherwise fall through
+                default:
+                    p = findParent(e);
+                    addProperty(p, e.name, e);
+                    break;
+            }
+        }
+
+        function annotate(e, path) {
+            e.setAttribute("path", path);
+            var o = e.formDom || {};
+            for (var key in o) {
+                var v = o[key];
+
+                function child(v, i) {
+                    var suffix = null;
+                    var newKey = key;
+                    if (v.parentNode.className && v.parentNode.className.indexOf("one-each") > -1 && v.parentNode.className.indexOf("honor-order") > -1) {
+                        suffix = v.getAttribute("descriptorId").split(".").pop()
+                    } else if (v.getAttribute("type") == "radio") {
+                        suffix = v.value
+                        while (newKey.substring(0, 8) == 'removeme')
+                            newKey = newKey.substring(newKey.indexOf('_', 8) + 1);
+                    } else if (v.getAttribute("suffix") != null) {
+                        suffix = v.getAttribute("suffix")
+                    } else {
+                        if (i > 0)
+                            suffix = i;
+                    }
+                    if (suffix == null) suffix = "";
+                    else suffix = '[' + suffix + ']';
+
+                    annotate(v, path + "/" + newKey + suffix);
+                }
+
+                if (v instanceof Array) {
+                    var i = 0;
+                    v.forEach(function (v) {
+                        child(v, i++)
+                    })
+                } else {
+                    child(v, 0)
+                }
+            }
+
+        }
+
+        annotate(form, "");
+
+        // clean up
+        for (i = 0; i < doms.length; i++)
+            doms[i].formDom = null;
+
+        return true;
+    }
+
+    function applyAll() {
+        document.querySelectorAll("FORM").forEach(function (e) {
+            buildFormTree(e);
+        })
+    }
+
+    /* JavaScript sometimes re-arranges the DOM and doesn't call layout callback
+     * known cases: YUI buttons, CodeMirror.
```

Comment:
Ahhhh!!!!    I wonder if this may well explain some issues we have with codemirror sections not being found in the ATH!